### PR TITLE
Challenge (for review)

### PR DESCRIPTION
**Pull request by TeToNN**

### CODE REVIEW

**Comment by NamTThai** [comment]
Can you pull into `updated-challenge` plz. I wanna run it first

File: app/src/main/res/layout/dialog_challenge.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    <Spinner
+        android:id="@+id/challenger_team_spinner"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_horizontal"
+        android:layout_margin="8dp" />
+
+
+    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (NamTThai) [inline app/src/main/res/layout/dialog_challenge.xml]: Just use AlertDialog for positive and negative button. You can use `setView` and `View.inflate()` to change AlertDialog view using the above TextView/Spinner

File: app/src/main/java/com/pokemonshowdown/data/BattleFieldData.java [hunk 2]
```diff
@@ -216,7 +216,6 @@ public void leaveAllRooms() {
             leaveRoom(roomId);
         }
     }
-
```
Comment (NamTThai) [inline app/src/main/java/com/pokemonshowdown/data/BattleFieldData.java]: No. lol

Comment (TeToNN) [inline app/src/main/java/com/pokemonshowdown/data/BattleFieldData.java]: didnt change anythng here?

File: app/src/main/res/layout/dialog_challenge.xml (hunk 1 above)
Comment (TeToNN) [inline app/src/main/res/layout/dialog_challenge.xml]: Alright

File: app/src/main/java/com/pokemonshowdown/data/BattleFieldData.java (hunk 2 above)
Comment (NamTThai) [inline app/src/main/java/com/pokemonshowdown/data/BattleFieldData.java]: You deleted a line

**Pull request closed without merging**